libafl/src/stages/power.rs

```diff
@@ -57,54 +80,296 @@ where
+        } else if q_bitmap_size * 0.75 > avg_bitmap_size as f64 {
+            perf_score *= 1.5;
+        } else if q_bitmap_size * 3.0 < avg_bitmap_size as f64 {
+            perf_score *= 0.25;
+        } else if q_bitmap_size * 2.0 < avg_bitmap_size as f64 {
+            perf_score *= 0.5;
+        } else if q_bitmap_size * 1.5 < avg_bitmap_size as f64 {
+            perf_score *= 0.75;
+        }
+
+        if metadata.handicap() >= 4 {
+            perf_score *= 4.0;
+            metadata.set_handicap(metadata.handicap() - 4);
+        } else if metadata.handicap() > 0 {
+            perf_score *= 2.0;
+            metadata.set_handicap(metadata.handicap() - 1);
+        }
+
+        if metadata.depth() >= 4 && metadata.depth() < 8 {
+            perf_score *= 2.0;
+        } else if metadata.depth() >= 8 && metadata.depth() < 14 {
+            perf_score *= 3.0;
+        } else if metadata.depth() >= 14 && metadata.depth() < 25 {
+            perf_score *= 4.0;
+        } else if metadata.depth() >= 25 {
+            perf_score *= 5.0;
+        }
+
+        let mut factor: f64 = 1.0;
+
+        // TODO: currently we don't have any favored inputs, if that's introduced we need to modify code here
+        match &self.strat {
+            PowerSchedule::EXPLORE => {
+                // Nothing happens in EXPLORE
+            }
+            PowerSchedule::EXPLOIT => {
+                factor = MAX_FACTOR;
+            }
+            PowerSchedule::COE => {
+                if f64::from(self.n_fuzz[metadata.n_fuzz_entry()]) > fuzz_mu {
+                    factor = 0.0;
+                }
+            }
+            PowerSchedule::FAST => {
+                if metadata.fuzz_level() != 0 {
+                    let lg = libm::log2(f64::from(self.n_fuzz[metadata.n_fuzz_entry()]));
+                    // Do thing if factor == 5
+                    if lg < 2.0 {
+                        factor = 4.0;
+                    } else if (2.0..4.0).contains(&lg) {
+                        factor = 3.0;
+                    } else if (4.0..5.0).contains(&lg) {
+                        factor = 2.0;
+                    } else if (6.0..7.0).contains(&lg) {
+                        factor = 0.8;
+                    } else if (7.0..8.0).contains(&lg) {
+                        factor = 0.6;
+                    } else if lg >= 8.0 {
+                        factor = 0.4;
+                    }
+                }
+            }
+            PowerSchedule::LIN => {
+                factor = (metadata.fuzz_level() as f64)
+                    / f64::from(self.n_fuzz[metadata.n_fuzz_entry()] + 1);
+            }
+            PowerSchedule::QUAD => {
+                factor = ((metadata.fuzz_level() * metadata.fuzz_level()) as f64)
+                    / f64::from(self.n_fuzz[metadata.n_fuzz_entry()] + 1);
+            }
+        }
+
+        perf_score *= factor / POWER_BETA;
+
+        // Lower bound if the strat is not COE.
+        match self.strat {
```

Comment:
Isn't that the same as
```rust
if self.strat == PowerScheduler::COE && perf_score < 1.0 { perf_score = 1.0; }
```
in which case I'd probably opt for the `if` (less verbose?)